common/types/ballot.go

```diff
@@ -204,7 +205,11 @@ type EpochData struct {
 
 // Initialize calculates and sets the Ballot's cached ballotID and smesherID.
 // this should be called once all the other fields of the Ballot are set.
-func (b *Ballot) Initialize() error {
+func (b *Ballot) Initialize(extractor keyExtractor) error {
```

Comment:
but why it needs to be passed like that? domain is already known in this function, in my opinion single line change would be much nicer

```go
ExtractNodeID(BallotDomain, data, sig)
``` 

or if you want to encapsulate BallotDomain in the object

```go
signing.Ballots.ExtractNodeID(data, sig)
```

Reply:
the original intent was to break the cyclic dependency btwn common/types and signing. i think i want to move ExtractKey outside of common/types pkg.